integrationtests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -348,35 +353,49 @@ private static void assertCreatingTopicThrowsExpectedException(Admin proxyAdmin)
 
     @Test
     void testCompositeFilter() {
-        try (MockServerKroxyliciousTester tester = mockKafkaKroxyliciousTester((mockBootstrap) -> proxy(mockBootstrap)
-                .addToFilters(new FilterDefinitionBuilder(CompositePrefixingFixedClientIdFilter.class.getName())
-                        .withConfig("clientId", "banana", "prefix", "123").build()));
+        final KroxyliciousTesterBuilder testerBuilder = new KroxyliciousTesterBuilder()
+                .addMockResponse(new ResponsePayload(METADATA, METADATA.latestVersion(), new MetadataResponseData()))
+                .setMockConfigurationFunction((mockBootstrap) -> proxy(mockBootstrap)
```

Comment:
I think setMockConfigurationFunction would baffle a newcomer.   Also, what would it mean if testBuilder.createMockKroxyliciousTester() were called without calling setMockConfigurationFunction first?

I'm wondering if KroxyliciousTesterBuilder() should expose a progressive API.  The caller would do

```java
KroxyliciousTesterBuilder().withMockUpstream(bootstrap -> proxy()..)
```

Reply:
Yes, I think I commented elsewhere that it was a smell, it's confusing which is one of the bigger points I was looking for input on before continuing.

I definitely think something progressive is the answer I'm just not clear what level it needs to be at as I didn't follow the usages through.

Validating that the setup makes sense is worth doing if we think the builder is exploring.

Reply:
I've moved the work touching the mock tester to https://github.com/kroxylicious/kroxylicious/pull/661 so I can keep this PR focused on the topicManager work